tests/check_coding_style.sh

```diff
@@ -0,0 +1,192 @@
+#!/bin/bash
+
+# check_coding_style.sh
+# Copyright (c) 2019 by Peter van der Meer
+#
+# Endless Sky is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+
+
+# This is a checker for coding style issues specifically aimed at following the
+# coding style as published on:
+# http://endless-sky.github.io/styleguide/styleguide.xml
+#
+# This checker performs heuristic pattern matching, so there is a small risk that
+# constructs will be rejected that are actually allowed. The white-list should be
+# used to deal with such false positives and can also be used for allowed deviations.
+
+
+# Determine path of the current script
+HERE=$(cd `dirname $0` && pwd)
+SOURCES="${HERE}/../source"
+
+# Desired header, adapted for automatic parsing
+DESIRED_HEADER="/* FILENAME
+Copyright (c) 2xxx by Michael Zahniser
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/"
+
+function die()
+{
+	echo "ERROR: $1"
+	exit 1
+}
+
+
+
+# This is a white-list of allowed exceptions. For now just included in the script;
+# might want to make this a separate text-file later.
+function filter_exceptions()
+{
+	grep -v \
+		-e "WinApp.rc" \
+		-e "Files.cpp:		struct stat buf;" \
+		-e "Files.cpp:	struct _stat buf;" \
+		-e "Files.cpp:	struct stat buf;" \
+		-e "ImageBuffer.cpp:		struct jpeg_error_mgr jerr;" \
+		-e "Point.h:		struct {" \
+		-e "#include <OpenGL/GL3.h>" \
+		-e "#include <GL/glew.h>" \
+		-e "#include <OpenAL/al.h>" \
+		-e "#include <OpenAL/alc.h>" \
+		-e "#include <SDL2/SDL.h>" \
+		-e "#include <SDL2/SDL_events.h>" \
+		-e "#include \"gl_header.h\"" \
+		-e "#include \"Information.h\"" \
+		-e "#include \"Rectangle.h\"" \
+		-e "#include \"Body.h\"" \
+		-e "#include <windows.h>" \
+		-e "#include <sys/stat.h>" \
+		-e "#include <dirent.h>" \
+		-e "#include <unistd.h>" \
+		-e "#include <mutex>" \
+		-e "#include <jpeglib.h>" \
+		-e "#include <png.h>" \
+		-e "#include <mad.h>" \
+		-e "#include \"Files.h\"" \
+		-e "main.cpp: mismatch in header" \
+		-e "Color.cpp:#include \"Color.h\"" \
+		-e "Rectangle.cpp:#include \"Rectangle.h\"" \
```

Comment:
How many of these exceptions should be kept vs just fixed?

Reply:
The struct ones really are an exception (the coding style recommends not using structs).
I moved WinApp.rc to the right location.
The #include checks were removed for performance reasons, so they no longer needed to be listed here (I removed them)
I fixed an issue for the using std which also removed an exception.
And I removed an exception regarding tabs/spaces but left the actual issue in the code to check/verify the style checking script.